content/docs/iac/concepts/plugins.md

```diff
@@ -0,0 +1,108 @@
+---
+title_tag: Plugins
+meta_desc: Learn about Pulumi plugins, the core extensibility mechanism that enables Pulumi to work with cloud providers, languages, and tools.
+title: Plugins
+h1: Plugins
+meta_image: /images/docs/meta-images/docs-meta.png
+menu:
+    iac:
+        name: Plugins
+        parent: iac-concepts
+        weight: 140
+    concepts:
+        weight: 140
+---
+
+{{% notes type="info" %}}
+Most Pulumi users don't need to understand plugins in depth, as most plugin installation and management happens automatically.
+
+If you're interested in learning more about Pulumi internals, see the [Pulumi Developer Documentation](https://pulumi-developer-docs.readthedocs.io/latest/docs/architecture/plugins.html).
+{{% /notes %}}
+
+Plugins are Pulumi's core extensibility mechanism, allowing the Pulumi engine to communicate in a uniform manner with various languages, resource providers, and other tools. Plugins typically run as separate processes and often use gRPC for communication with the Pulumi engine.
+
+## Plugin types
+
+Pulumi supports five categories of plugins:
+
+### Resource plugins
+
+Resource plugins (also known as [providers](/docs/iac/concepts/resources/providers/)) expose standardized interfaces for managing cloud resources. A listing of providers is available in the [Pulumi Registry](/registry/).
+
+When you install a provider SDK package (for example, via `npm install @pulumi/aws` in Node.js), the package manager automatically downloads and installs the provider plugin executable along with the SDK if it's not already cached on your system.
```

Comment:
"the package manager automatically downloads" pretty sure this isn't true anymore, it was once for nodejs and python but we now just let the engine download all resource plugins on first `up/preview` or `install`

Reply:
You are correct and this is now fixed.